jetty/src/main/scala/org/http4s/server/jetty/JettyBuilder.scala

```diff
@@ -13,17 +13,16 @@ import org.eclipse.jetty.servlet.{FilterHolder, ServletContextHandler, ServletHo
 import org.eclipse.jetty.util.component.AbstractLifeCycle.AbstractLifeCycleListener
 import org.eclipse.jetty.util.component.LifeCycle
 import org.eclipse.jetty.util.ssl.SslContextFactory
-import org.eclipse.jetty.util.thread.QueuedThreadPool
+import org.eclipse.jetty.util.thread.{QueuedThreadPool, ThreadPool}
 import org.http4s.server.SSLKeyStoreSupport.StoreInfo
 import org.http4s.servlet.{AsyncHttp4sServlet, ServletContainer, ServletIo}
 import org.log4s.getLogger
-import scala.concurrent.ExecutionContext
 import scala.collection.immutable
 import scala.concurrent.duration._
 
 sealed class JettyBuilder[F[_]] private (
     socketAddress: InetSocketAddress,
-    private val executionContext: ExecutionContext,
+    threadPool: ThreadPool,
```

Comment:
Actually, that's probably not true: we've relinquished that servlet thread and it's uncertain what shifting back onto it would do to the server's internal bookkeeping.

In fact, blocking IO in the blocking servlet shouldn't shift at all, because the servlet thread will be waiting anyway.  The only time this shifting is useful at all is blocking IO in the async servlet, which itself is only useful on 3.0 -- not 2.5 (blocking everywhere) or 3.1 (non-blocking servlet IO).

Greetings from 2010.